### ISSUE TITLE
Demo.ipynb not working : *** No instances to display ***

### ISSUE THREAD

**Issue by tobymu** [matterport/Mask_RCNN#942]
@waleedka Hello, I use tensorflow1.9 + python3.5

When I run the demo, it shows "No instances to display"

`import os
import sys
import random
import math
import numpy as np
import skimage.io
import matplotlib
import matplotlib.pyplot as plt

# Root directory of the project
ROOT_DIR = os.path.abspath("..")
print("ROOT_DIR:", ROOT_DIR)

# Import Mask RCNN
sys.path.append(ROOT_DIR)  # To find local version of the library
from mrcnn import utils
import mrcnn.model as modellib
from mrcnn import visualize
# Import COCO config
sys.path.append(os.path.join(ROOT_DIR, "samples/coco/"))  # To find local version
import coco

# %matplotlib inline 

# Directory to save logs and trained model
MODEL_DIR = os.path.join(ROOT_DIR, "logs")

# Local path to trained weights file
COCO_MODEL_PATH = os.path.join(ROOT_DIR, "mask_rcnn_coco.h5")
print("COCO_MODEL_PATH", COCO_MODEL_PATH)
# Download COCO trained weights from Releases if needed
if not os.path.exists(COCO_MODEL_PATH):
    utils.download_trained_weights(COCO_MODEL_PATH)

# Directory of images to run detection on
IMAGE_DIR = os.path.join(ROOT_DIR, "images")
print("IMAGE_DIR", IMAGE_DIR)

class InferenceConfig(coco.CocoConfig):
    # Set batch size to 1 since we'll be running inference on
    # one image at a time. Batch size = GPU_COUNT * IMAGES_PER_GPU
    GPU_COUNT = 1
    IMAGES_PER_GPU = 1

config = InferenceConfig()
# config.display()

# Create model object in inference mode.
model = modellib.MaskRCNN(mode="inference", model_dir=MODEL_DIR, config=config)

# Load weights trained on MS-COCO
model.load_weights(COCO_MODEL_PATH, by_name=True)

# COCO Class names
# Index of the class in the list is its ID. For example, to get ID of
# the teddy bear class, use: class_names.index('teddy bear')
class_names = ['BG', 'person', 'bicycle', 'car', 'motorcycle', 'airplane',
               'bus', 'train', 'truck', 'boat', 'traffic light',
               'fire hydrant', 'stop sign', 'parking meter', 'bench', 'bird',
               'cat', 'dog', 'horse', 'sheep', 'cow', 'elephant', 'bear',
               'zebra', 'giraffe', 'backpack', 'umbrella', 'handbag', 'tie',
               'suitcase', 'frisbee', 'skis', 'snowboard', 'sports ball',
               'kite', 'baseball bat', 'baseball glove', 'skateboard',
               'surfboard', 'tennis racket', 'bottle', 'wine glass', 'cup',
               'fork', 'knife', 'spoon', 'bowl', 'banana', 'apple',
               'sandwich', 'orange', 'broccoli', 'carrot', 'hot dog', 'pizza',
               'donut', 'cake', 'chair', 'couch', 'potted plant', 'bed',
               'dining table', 'toilet', 'tv', 'laptop', 'mouse', 'remote',
               'keyboard', 'cell phone', 'microwave', 'oven', 'toaster',
               'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors',
               'teddy bear', 'hair drier', 'toothbrush']

# Load a random image from the images folder
file_names = next(os.walk(IMAGE_DIR))[2]
print(file_names)
image = skimage.io.imread(os.path.join(IMAGE_DIR, random.choice(file_names)))

# Run detection
results = model.detect([image], verbose=1)

# Visualize results
r = results[0]
visualize.display_instances(image, r['rois'], r['masks'], r['class_ids'], 
                            class_names, r['scores'])`

I don't know why. Thanks for help!

**Comment by tabelh** [comment]
Any news on this subject, running in to the same subject. In my case it is caused by a divide by zero in model.py, 2456: boxes = np.divide(boxes - shift, scale) using tensorflow 1.13 python 2.7 running demo.ipynb in jupyter.

**Comment by musematics** [comment]
> Any news on this subject, running in to the same problem. In my case it is caused by a divide by zero in model.py, 2456: boxes = np.divide(boxes - shift, scale) using tensorflow 1.13 python 2.7 running demo.ipynb in jupyter.
@tabelh @tobymu 
I encountered exactly the same problem here. Any solution? Thanks al lot!

**Comment by tabelh** [comment]
> > Any news on this subject, running in to the same problem. In my case it is caused by a divide by zero in model.py, 2456: boxes = np.divide(boxes - shift, scale) using tensorflow 1.13 python 2.7 running demo.ipynb in jupyter.
> 
> @tabelh @tobymu
> I encountered exactly the same problem here. Any solution? Thanks al lot!

I think it has something to do with the interpreter of python2.7 that does not allow simple training. I ended up simply running the code in a python3 virtualenvironment to do my training. The weight file can then be used for inference with python2 without issues, at least i haven't found them.

**Comment by musematics** [comment]
> Any news on this subject, running in to the same problem. In my case it is caused by a divide by zero in model.py, 2456: boxes = np.divide(boxes - shift, scale) using tensorflow 1.13 python 2.7 running demo.ipynb in jupyter.
> 
> @tabelh @tobymu
> I encountered exactly the same problem here. Any solution? Thanks al lot!
> 
> I think it has something to do with the interpreter of python2.7 that does not allow simple training. I ended up simply running the code in a python3 virtualenvironment to do my training. The weight file can then be used for inference with python2 without issues, at least i haven't found them.

@tabelh 
Thank you very much! I've just found 'Working with py2 [#515](https://github.com/matterport/Mask_RCNN/pull/515). A modification with model.py and visualize.py might also solve this problem.
